developer_manual/digging_deeper/dashboard.rst

```diff
@@ -0,0 +1,176 @@
+=========
+Dashboard
+=========
+
+The dashboard app aims to provide the user with a general overview of their
+Nextcloud and shows information that is currently important.
+
+App developers can integrate into the dashboard app and provide their own panels.
+
+
+Register a dashboard panels
+---------------------------
+
+A dashboard panel is represented by a class implementing the `OCP\\Dashboard\\IPanel`
+interface. This class will be instantiated whenever the dashboard is loaded.
+Any bootstrap code that is needed for the panel can be implemented inside
+of the `load` method and will be called when the dashboard is loaded.
+
+.. code-block:: php
+
+    <?php
+
+    namespace OCA\MyApp\Dashboard;
+
+    use OCP\Dashboard\IPanel;
+    use OCP\IInitialStateService;
+    use OCP\IL10N;
+    use OCP\IURLGenerator;
+    use OCP\IUserSession;
+
+    class MyAppPanel implements IPanel {
+
+        public function __construct(
+            IInitialStateService $initialStateService,
+            IL10N $l10n,
+            IURLGenerator $urlGenerator
+        ) {
+            $this->initialStateService = $initialStateService;
+            $this->l10n = $l10n;
+            $this->urlGenerator = $urlGenerator;
+        }
+
+        /**
+         * @return string Unique id that identifies the panel, e.g. the app id
+         * @since 20.0.0
+         */        public function getId(): string {
+            return 'myapppanelid';
+        }
+
+        /**
+         * @return string User facing title of the panel
+         * @since 20.0.0
+         */
+        public function getTitle(): string {
+            return $this->l10n->t('My app');
+        }
+
+        /**
+         * @return int Initial order for panel sorting
+         * @since 20.0.0
+         */
+        public function getOrder(): int {
+            return 0;
+        }
+
+        /**
+         * @return string css class that displays an icon next to the panel title
+         * @since 20.0.0
+         */
+        public function getIconClass(): string {
+            return 'icon-class';
+        }
+
+        /**
+         * @return string|null The absolute url to the apps own view
```

Comment:
what about apps that don't have this? should we consider those and make this nullable?

Reply:
Fixed, I forgot to also add the nullable here, it is already in place in the interface:
https://github.com/nextcloud/server/blob/ea46d1a3a2e99b77cb35dd5eefa895406cbceb58/lib/public/Dashboard/IPanel.php#L62